aten/src/ATen/native/native_functions.yaml

```diff
@@ -14726,6 +14726,13 @@
   dispatch:
     CompositeExplicitAutograd: wait_tensor
 
+- func: all_reduce_coalesced(Tensor[] self, str reduceOp, str tag, int[] ranks, int group_size) -> Tensor[]
```

Comment:
I know that this might not be ready for review, but would like to have some suggestion about the naming. Given that we are in the aten op world, shall we make this an overload of `all_reduce` op instead of making them separate ops? i.e. I feel we can do `all_reduce.coalesced` or `all_reduce.grouped`

Reply:
IDK, this will break a lot of our existing passes as they don't expect different ops with the same name.